catalyst-toolbox/src/bin/cli/rewards/voters.rs

```diff
@@ -65,30 +68,51 @@ impl VotersRewards {
             votes_count_path,
             vote_threshold,
         } = self;
-        let block = common.input.load_block()?;
-        let block0 = Block0Configuration::from_block(&block)?;
-
-        let vote_count: VoteCount = serde_json::from_reader(jcli_lib::utils::io::open_file_read(
-            &Some(votes_count_path),
-        )?)?;
-
-        let snapshot = Snapshot::from_raw_snapshot(
-            serde_json::from_reader(jcli_lib::utils::io::open_file_read(&Some(snapshot_path))?)?,
-            registration_threshold.into(),
-        );
 
-        let results = calc_voter_rewards(
-            vote_count,
+        voter_rewards(
+            common
+                .output_file
+                .as_deref()
+                .ok_or(eyre!("missing block file"))?,
+            votes_count_path,
+            snapshot_path,
+            registration_threshold,
             vote_threshold,
-            &block0,
-            snapshot,
-            Rewards::from(total_rewards),
-        )?;
-
-        let actual_rewards = results.values().sum::<Rewards>();
-        assert_are_close(actual_rewards, Rewards::from(total_rewards));
-
-        write_rewards_results(common, &results)?;
-        Ok(())
+            total_rewards,
+        )
     }
 }
+
+pub fn voter_rewards(
+    block_file: &Path,
+    votes_count_path: PathBuf,
+    snapshot_path: PathBuf,
+    registration_threshold: u64,
+    vote_threshold: u64,
+    total_rewards: u64,
+) -> Result<()> {
+    let block = open_block_file(&Some(block_file.to_path_buf()))?;
+    let block = load_block(block)?;
+
+    let block0 = Block0Configuration::from_block(&block)?;
+
+    let vote_count: VoteCount = serde_json::from_reader(open_file_read(&Some(votes_count_path))?)?;
+    let snapshot = Snapshot::from_raw_snapshot(
+        serde_json::from_reader(open_file_read(&Some(snapshot_path))?)?,
+        registration_threshold.into(),
+    );
+
+    let results = calc_voter_rewards(
+        vote_count,
+        vote_threshold,
+        &block0,
+        snapshot,
+        Rewards::from(total_rewards),
+    )?;
+
+    let actual_rewards = results.values().sum::<Rewards>();
+    assert_are_close(actual_rewards, Rewards::from(total_rewards));
+
+    write_rewards_results(&Some(block_file.to_path_buf()), &results)?;
```

Comment:
why writing to block0? this should have its own output file

Reply:
I got mixed up when splitting this code out, I didn't see that it was using `common.output`